api/app/Observers/UserObserver.php

```diff
@@ -16,14 +17,16 @@ public function created(User $user): void
 
     public function updating(User $user)
     {
-        if ($user->isDirty(['wfa_date', 'wfa_interest'])) {
-            $newInterest = $user->wfa_interest;
+        $interestDirty = $user->isDirty('wfa_interest');
+        $wfaDateDirty = $user->isDirty('wfa_date');
 
-            if (is_null($newInterest)) {
-                $user->wfa_date = null;
-            }
+        if ($interestDirty || $wfaDateDirty) {
             $user->wfa_updated_at = now();
         }
+
+        if (is_null($user->wfa_interest) || $user->wfa_interest === WfaInterest::NOT_APPLICABLE->name) {
+            $user->wfa_date = null;
+        }
     }
```

Comment:
Voluntary departure should probably null this out too

Reply:
Is this correct @Jerryescandon ?